### PR TITLE
Validate PATCH /line/:lineid/bounds requests

### LINKED ISSUE #342: Raw User Input: PATCH /line/:lineId/bounds route
The pull request generated to contribute the work should target the 'development' branch.

/line/index.js has the following registered endpoint.

```javascript
router.route("/:lineId/bounds").patch(...)
```
It does not have any route protection for the project data provided by the client in the request body.  It should use the middleware from /utilities/checkIfSuspicious.js to protect the route.

### LINKED ISSUE #355: Validate PATCH /line/:lineid/bounds request bodies
The body is already validated as an object with x,y,w, and h properties.  However, whatever values those properties have are passed through and used, even through the class.  They should be validated as integers or strings that can be integers, otherwise malicious strings can be passed through and end up in HTML.

See `router.patch('/:lineId/bounds')` and `Line.updateBounds()`

### LINKED ISSUE #432: updateBounds() logic error
```javascript
async updateBounds({x, y, w, h}, options = {}) {
    if (!x || !y || !w || !h) {
        throw new Error('Bounds ({x,y,w,h}) must be provided')
    }
    if (options.creator) this.creator = options.creator
    this.target ??= ''
    const newTarget = this.updateTargetXYWH(this.target, x, y, w, h)
    if (this.target === newTarget) {
        return this
    }
    this.target = newTarget
    return this.update()
}
```

Yeah but I made the top left corner (0, 0) and you don't support 0.  x=0 and y=0 should be valid bounds.

### PR DESCRIPTION
**Pull request by thehabes**
Closes #342 
Closes #355
Closes #432 

## Summary
- Replaced loose falsy checks (`!x || !y || !w || !h`) with strict validation requiring non-negative integers or digit-only strings
- Added `parseInt()` coercion so validated string values (e.g. `"100"`) are converted to integers before reaching `updateTargetXYWH()`
- Added `null` and `Array` body guards in the route handler to return a proper 400 instead of 500
- Applied validation defense-in-depth at both the route handler (`line/index.js`) and domain model (`classes/Line/Line.js`)

## What was wrong
The `PATCH /:lineId/bounds` endpoint and `Line.updateBounds()` accepted arbitrary values for `x`, `y`, `w`, and `h` properties. Improper or malicious value strings could be passed through and interpolated into IIIF fragment selectors (`#xywh=pixel:${x},${y},${w},${h}`), potentially ending up in HTML.

## What changed
- `isValidBound` checks `Number.isInteger(v) && v >= 0` for integers and `/^\d+$/.test(v)` for string representations
- Validated values are coerced to integers via `parseInt(v, 10)` before any downstream use
- Route-level guards added for `null` body and `Array` body edge cases

## Test plan
- [ ] Valid integer bounds (`{ x: 20, y: 20, w: 200, h: 200 }`) accepted as before
- [ ] Valid string integer bounds (`{ x: "20", y: "20", w: "200", h: "200" }`) accepted and coerced
- [ ] Zero origin (`{ x: 0, y: 0, w: 100, h: 100 }`) now accepted (previously rejected as falsy)
- [ ] Malicious strings (`{ x: "<script>alert(1)</script>", ... }`) rejected with 400
- [ ] Negative values (`{ x: -1, ... }`) rejected with 400
- [ ] Float values (`{ x: 1.5, ... }`) rejected with 400
- [ ] `null` JSON body returns 400, not 500

### CODE REVIEW

File: line/index.js [hunk 1]
```diff
@@ -270,9 +270,11 @@ router.patch('/:lineId/bounds', auth0Middleware(), async (req, res) => {
     if (!(await projectObj.checkUserAccess(user._id, ACTIONS.UPDATE, SCOPES.SELECTOR, ENTITIES.LINE))) {
       return respondWithError(res, 403, 'You do not have permission to update line bounds in this project')
     }
-    if (typeof req.body !== 'object' || !req.body.x || !req.body.y || !req.body.w || !req.body.h) {
-      return respondWithError(res, 400, 'Invalid request body. Expected an object with x, y, w, and h properties.')
+    const isValidBound = v => (Number.isInteger(v) && v >= 0) || (typeof v === 'string' && /^\d+$/.test(v))
+    if (!req.body || typeof req.body !== 'object' || Array.isArray(req.body) || !isValidBound(req.body.x) || !isValidBound(req.body.y) || !isValidBound(req.body.w) || !isValidBound(req.body.h)) {
```
[the remark this answers is not in the document]
Comment (thehabes) [reply line/index.js]: Could be if there if there is a good way to change it.  isValidBound() is just checking if the coordinate is valid or not, so it should not be changing anything which is typically what one uses `map()` for.  It was a bigger rewrite in this area to do that tho.

File: classes/Line/Line.js [hunk 2]
```diff
@@ -180,9 +180,11 @@ export default class Line {
     }
 
     async updateBounds({x, y, w, h}, options = {}) {
-        if (!x || !y || !w || !h) {
-            throw new Error('Bounds ({x,y,w,h}) must be provided')
+        const isValidBound = v => (Number.isInteger(v) && v >= 0) || (typeof v === 'string' && /^\d+$/.test(v))
+        if (!isValidBound(x) || !isValidBound(y) || !isValidBound(w) || !isValidBound(h)) {
+            throw new Error('Bounds ({x,y,w,h}) must be non-negative integers')
```
[the remark this answers is not in the document]
Comment (thehabes) [reply classes/Line/Line.js]: Yes, this was written to only do rectangular bounds noted by integer-only selections.  We would have to write a different one to do different selectors.
